src/test/regress/sql/explain_analyze.sql

```diff
@@ -37,3 +49,29 @@ select gp_inject_fault('explain_analyze_sort_error', 'reset', dbid)
     from gp_segment_configuration where role = 'p' and content > -1;
 drop table sort_error_test1;
 drop table sort_error_test2;
+
+--
+-- Test correct slice stats reporting in case of duplicated subplans
+--
+
+create table slice_test(i int, j int) distributed by (i);
+create table slice_test2(i int, j int) distributed by (i);
+insert into slice_test select i, i from generate_series(0, 100) i;
+insert into slice_test select i, 2*i from generate_series(0, 100) i;
+insert into slice_test2 values (0, 1);
+
+-- explain_processing_off
+-- create duplicate subplan in QE slice
+explain (analyze, timing off, costs off) select a.i from slice_test a
+  where a.j = (select b.i from slice_test2 b where a.i = 0 or b.i = 0)
+    and a.i = a.j;
+
+-- create multiple initplans in slice 0 (should be printed as two slices)
+explain (analyze, timing off, costs off)
+  select a.i from (select x::int as i, x::int / 5 as j from round(random() / 5) as x) a
+    where a.j = (select round(random() / 5)::int where a.i = 0)
+      and a.i = a.j;
```

Comment:
Is it expected that this test passes without the patch?

Reply:
Yes, the patch is supposed to fix the first query. The second query is there to make sure the patch doesn't break the current behavior with multiple slice 0 initplans.

Reply:
> The second query is there to make sure the patch doesn't break the current behavior with multiple slice 0 initplans.

It would be good to reflect this in the comments.